app/services/settings-v2/video.ts

```diff
@@ -3,80 +3,201 @@ import { Inject } from 'services/core/injector';
 
   initialState = {
-    videoContext: null as obs.IVideo,
+    default: null as obs.IVideoInfo,
+    horizontal: null as obs.IVideoInfo,
+    green: null as obs.IVideoInfo,
   };
 
   init() {
     this.establishVideoContext();
+    if (this.greenService.views.activeDisplays.green) {
+      this.establishVideoContext('green');
+    }
   }
 
+  contexts = {
+    default: null as obs.IVideo,
+    horizontal: null as obs.IVideo,
+    green: null as obs.IVideo,
+  };
+
   get videoSettingsValues() {
-    const context = this.state.videoContext;
+    const context = this.contexts.horizontal;
+    return {
+      baseRes: `${context.video.baseWidth}x${context.video.baseHeight}`,
+      outputRes: `${context.video.outputWidth}x${context.video.outputHeight}`,
+      scaleType: context.video.scaleType,
+      fpsType: context.video.fpsType,
+      fpsCom: `${context.video.fpsNum}-${context.video.fpsDen}`,
+      fpsNum: context.video.fpsNum,
+      fpsDen: context.video.fpsDen,
+      fpsInt: context.video.fpsNum,
+    };
+  }
+
+  get baseResolution() {
+    const context = this.contexts.horizontal;
     return {
-      baseRes: `${context.baseWidth}x${context.baseHeight}`,
-      outputRes: `${context.outputWidth}x${context.outputHeight}`,
-      scaleType: context.scaleType,
-      fpsType: context.fpsType,
-      fpsCom: `${context.fpsNum}-${context.fpsDen}`,
-      fpsNum: context.fpsNum,
-      fpsDen: context.fpsDen,
-      fpsInt: context.fpsNum,
+      width: context.video.baseWidth,
+      height: context.video.baseHeight,
     };
   }
 
+  get isGreen(): boolean {
+    return false;
+  }
+
   get videoSettings() {
-    return this.settingsManagerService.videoSettings;
+    return this.greenService.views.videoSettings;
   }
 
-  migrateSettings() {
-    Object.keys(this.videoSettings).forEach((key: keyof obs.IVideo) => {
-      this.SET_VIDEO_SETTING(key, this.videoSettings[key]);
-    });
+  migrateSettings(display?: TDisplayType) {
+    if (display === 'default' || display === 'horizontal') {
+      this.state.horizontal = this.contexts.horizontal.video;
+      Object.keys(this.contexts.horizontal.legacySettings).forEach(
+        (key: keyof obs.IAdvancedStreaming | keyof obs.ISimpleStreaming) => {
+          this.SET_VIDEO_SETTING(key, this.contexts.horizontal.legacySettings[key]);
+        },
+      );
+      Object.keys(this.contexts.horizontal.video).forEach((key: keyof obs.IVideo) => {
+        this.SET_VIDEO_SETTING(key, this.contexts.horizontal.video[key]);
+        if (
+          invalidFps(this.contexts.horizontal.video.fpsNum, this.contexts.horizontal.video.fpsDen)
+        ) {
+          this.createDefaultFps();
+        }
+      });
+    } else {
+      const data = this.videoSettings.green;
 
-    if (invalidFps(this.state.videoContext.fpsNum, this.state.videoContext.fpsDen)) {
-      this.setVideoSetting('fpsNum', 30);
-      this.setVideoSetting('fpsDen', 1);
+      this.state[display] = this.contexts[display].video;
```

Comment:
we shouldn't set state outside of mutations

Reply:
Done